pkg/kubectl/cmd/version.go

```diff
@@ -46,36 +54,74 @@ func NewCmdVersion(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	}
 	cmd.Flags().BoolP("client", "c", false, "Client version only (no server required).")
 	cmd.Flags().BoolP("short", "", false, "Print just the version number.")
+	cmd.Flags().String("output", "", "output format, options available are yaml and json")
 	cmd.Flags().MarkShorthandDeprecated("client", "please use --client instead.")
 	return cmd
 }
 
 func RunVersion(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
-	v := fmt.Sprintf("%#v", version.Get())
-	if cmdutil.GetFlagBool(cmd, "short") {
-		v = version.Get().GitVersion
+	var serverVersion *apimachineryversion.Info = nil
+	var serverErr error = nil
+	vo := VersionObj{nil, nil}
+
+	clientVersion := version.Get()
+	vo.ClientVersion = &clientVersion
+
+	if !cmdutil.GetFlagBool(cmd, "client") {
+		serverVersion, serverErr = retrieveServerVersion(f)
+		vo.ServerVersion = serverVersion
 	}
 
-	fmt.Fprintf(out, "Client Version: %s\n", v)
-	if cmdutil.GetFlagBool(cmd, "client") {
-		return nil
+	switch of := cmdutil.GetFlagString(cmd, "output"); of {
+	case "":
+		if cmdutil.GetFlagBool(cmd, "short") {
```

Comment:
I dont think there would be much gained by adding an error if both flags are used. The suggestion goes back to my disagreement regarding expected/unexpected behavior using flags i had prior, please look at the history, regarding how it is implemented now. If you feel strongly about this, I think it would be best we get together and discuss it in detail and perhaps discuss some enforcement throughout the code, or what exists atm.